### PR TITLE
Add support for LPC USB

### PR DESCRIPTION
**Pull request by mmahadevan108**

### CODE REVIEW

**Review by galak — COMMENTED** [review]
Can we take this chance to move the zephyr code into zephyr?

**Comment by galak** [comment]
> Can we take this chance to move the zephyr code into zephyr?

To add more details, in the zephyr code base we'd have this in `modules/hal_nxp` (see how `modules/hal_nordic` for example).

**Comment by mmahadevan108** [comment]
> > Can we take this chance to move the zephyr code into zephyr?
> 
> To add more details, in the zephyr code base we'd have this in `modules/hal_nxp` (see how `modules/hal_nordic` for example).

I have created a separate PR for this. 
https://github.com/zephyrproject-rtos/zephyr/pull/37215
https://github.com/zephyrproject-rtos/hal_nxp/pull/104

However I am seeing build issues even though I have added the header file to the include path. Not sure what I am missing. @galak @MaureenHelm, can you please take a look and provide your inputs.

**Comment by galak** [comment]
> [](https://github.com/zephyrproject-rtos/hal_nxp/pull/104)
> 
> However I am seeing build issues even though I have added the header file to the include path. Not sure what I am missing. @galak @MaureenHelm, can you please take a look and provide your inputs.

There is some zephyr module magic needed.  I pushed a set of PRs to put in the module magic that you should be able to build upon.

**Comment by mmahadevan108** [comment]
I have rebased this PR to get the SDK 2.10 update.

File: mcux/middleware/usb/device/usb_device_lpcip3511.c [hunk 1]
```diff
@@ -0,0 +1,2335 @@
+/*
+ * Copyright (c) 2016, Freescale Semiconductor, Inc.
+ * Copyright 2016 - 2017,2019 NXP
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "usb_dc_mcux.h"
```
Comment (MaureenHelm) [inline mcux/middleware/usb/device/usb_device_lpcip3511.c:9]: Please update the commit message to specify which version of the SDK the LPC USB driver comes from

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: mcux/middleware/usb/device/usb_device_lpcip3511.c (hunk 1 above)
Comment (mmahadevan108) [reply mcux/middleware/usb/device/usb_device_lpcip3511.c:9]: Done

File: mcux/middleware/usb/device/usb_device_ehci.c [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-#define DT_DRV_COMPAT nxp_kinetis_usbd
+#define DT_DRV_COMPAT nxp_mcux_usbd
```
Comment (dleach02) [inline mcux/middleware/usb/device/usb_device_ehci.c:10]: Was this driver initially used for Kinetis based USBD?

**Review by dleach02 — APPROVED** [review]

File: mcux/middleware/usb/device/usb_device_ehci.c (hunk 2 above)
Comment (mmahadevan108) [reply mcux/middleware/usb/device/usb_device_ehci.c:10]: No, it was used for MXRT devices only.

**Comment by mmahadevan108** [comment]
Anything else needed for this PR?

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**